operators/derive/to_dp.py

```diff
@@ -0,0 +1,337 @@
+                'service_urls',
+                'response_key',
+                'situation_key',
+                'organization_key',
+                'branch_key',
+            ],
+            resources=['flat_services'],
+        ),
+        DF.validate(),
+        DF.dump_to_path(f'{settings.DATA_DUMP_DIR}/flat_services'),
+    )
+
+
+def table_data_flow():
+    """Produce a table to back our Data APIs."""
+
+    return DF.Flow(
+        DF.load(
+            f'{settings.DATA_DUMP_DIR}/srm_data/datapackage.json',
+            resources=['responses', 'situations', 'organizations'],
+        ),
+        DF.load(
+            f'{settings.DATA_DUMP_DIR}/flat_branches/datapackage.json',
+            resources=['flat_branches'],
+        ),
+        DF.load(
+            f'{settings.DATA_DUMP_DIR}/flat_services/datapackage.json',
+            resources=['flat_services'],
+        ),
+        DF.update_package(name='Table Data'),
+        DF.update_resource(['flat_services'], name='table_data', path='table_data.csv'),
+        DF.join(
+            'responses',
+            ['key'],
+            'table_data',
+            ['response_key'],
+            fields=dict(
+                response_id={'name': 'id'},
+                response_name={'name': 'name'},
+            ),
+        ),
+        DF.join(
+            'situations',
+            ['key'],
+            'table_data',
+            ['situation_key'],
+            fields=dict(
+                situation_id={'name': 'id'},
+                situation_name={'name': 'name'},
+            ),
+        ),
+        DF.join(
+            'organizations',
+            ['key'],
+            'table_data',
+            ['organization_key'],
+            fields=dict(
+                organization_id={'name': 'id'},
+                organization_name={'name': 'name'},
+            ),
+        ),
+        DF.join(
+            'flat_branches',
+            ['branch_key'],
+            'table_data',
+            ['branch_key'],
+            fields=dict(
+                branch_id=None,
+                branch_name=None,
+                branch_geometry=None,
+                branch_address=None,
+            ),
+        ),
+        DF.add_field(
+            'response_parent_id',
+            'string',
+            lambda r: ":".join(r['response_id'].split(':')[:-1]),
+            resources=['table_data'],
+        ),
+        DF.select_fields(
+            [
+                # TODO: do we want to push these forward for data provenance?
+                # May be useful for future debugging, maybe not.
+                # 'service_key',
+                # 'response_key',
+                # 'situation_key',
+                # 'organization_key',
+                # 'branch_key',
+                'service_id',
+                'service_name',
+                'response_id',
+                'response_name',
+                'response_parent_id',
+                'situation_id',
```

Comment:
what about `branch_situation_id` and `organization_situation_id` - they seem to get lost in the process?

Reply:
Yes.